File: Java/Working with Numbers/53. Find roots of a Quadratic equation.java
import java.util.Scanner;

class Main {
    public static void main(String [] args) {
        Scanner userInput = new Scanner(System.in);

        System.out.println("Enter coefficients for ax^2 + bx + c = 0");
        System.out.print("Enter coefficient a: ");
        double a = userInput.nextDouble();
        System.out.print("Enter coefficient b: ");
        double b = userInput.nextDouble();
        System.out.print("Enter coefficient c: ");
        double c = userInput.nextDouble();

        double determinant = (b * b) - (4 * a * c);

        if (determinant > 0) {
            double root1 = (-b + Math.sqrt(determinant)) / (2 * a);
            double root2 = (-b - Math.sqrt(determinant)) / (2 * a);
            System.out.printf("Roots are Real and Different: %.2f and %.2f\n", root1, root2);
        } 
        else if (determinant == 0) {
            double root = -b / (2 * a);
            System.out.printf("Roots are Real and Equal: %.2f\n", root);
        } 
        else {
            double realPart = -b / (2 * a);
            double imaginaryPart = Math.sqrt(-determinant) / (2 * a);
            System.out.printf("Roots are Complex and Imaginary: %.2f + %.2fi and %.2f - %.2fi\n", realPart, imaginaryPart, realPart, imaginaryPart);
        }
        userInput.close();
    }
}

File: Python/Working with Numbers/53. Find roots of a Quadratic equation.py
print("Enter coefficients for ax^2 + bx + c = 0")
a = int(input("Enter the value of A :"))
b = int(input("Enter the value of B :"))
c = int(input("Enter the value of C :"))

d = ( -b + ((b ** 2) - 4 * a * c ) ** 0.5 ) / (2 * a)
e = ( -b - ((b ** 2) - 4 * a * c ) ** 0.5 ) / (2 * a)
f = (b ** 2) - 4 * a * c

if (f > 0):
    print("The roots are Real and Different")
    print("The roots are:", "Root 1:",d,"Root 2:", e)
elif (f == 0):
    print("The roots are Real and Equal")
    print("The roots are:", e)
else:
    print("The roots are Imaginary and Complex")